lib/rules/no-global-regexp-flag-in-query.ts

```diff
@@ -0,0 +1,83 @@
+import { TSESTree } from '@typescript-eslint/utils';
+
+import { createTestingLibraryRule } from '../create-testing-library-rule';
+import {
+  isMemberExpression,
+  isIdentifier,
+  isCallExpression,
+  isProperty,
+  isObjectExpression,
+} from '../node-utils';
+
+export const RULE_NAME = 'no-global-regexp-flag-in-query';
+export type MessageIds = 'noGlobalRegExpFlagInQuery';
+type Options = [];
+
+export default createTestingLibraryRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallow the use of the global RegExp flag (/g) in queries',
+      recommendedConfig: {
+        dom: 'error',
+        angular: 'error',
+        react: 'error',
+        vue: 'error',
```

Comment:
Per our [Semantic Versioning Policy](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/semantic-versioning-policy.md), reporting more errors from our recommended config would be a major version. Let's avoid setting this rule in any recommended config for now, and we will add it to the list of changes for the next major of the plugin.